File: mobile/src/androidTest/groovy/com/pablisco/android/estimate/EstimateApplicationTest.java
package com.pablisco.android.estimate;

import android.test.ApplicationTestCase;

import com.pablisco.android.estimate.graph.Graph;
import com.pablisco.reflect.ReflectiveType;

import static org.assertj.core.api.Assertions.assertThat;

public class EstimateApplicationTest extends ApplicationTestCase<EstimateApplication> {

    public EstimateApplicationTest() {
        super(EstimateApplication.class);
    }

    public void testInitializeGraph() throws Exception {
        createApplication();

        Graph actualGraph = ReflectiveType.reflect(Graph.graph()).field("GRAPH").getAs(Graph.class);

        assertThat(actualGraph).isNotNull();

    }
}

File: mobile/src/test/kotlin/com/pablisco/android/estimate/EstimateApplicationTest.kt
package com.pablisco.android.estimate

import com.pablisco.android.estimate.graph.Graph
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class EstimateApplicationTest {

    @Test fun shouldInitializeGraphOnCreate() {
        val application = EstimateApplication()
        application.onCreate()
        assertThat(Graph.graph()).isNotNull()
    }

}
